src/commands/tileindex-validate/tileindex.validate.ts

```diff
@@ -240,6 +251,18 @@ export const commandTileIndexValidate = command({
     if (projections.size > 1) {
       logger.warn({ projections: [...projections] }, 'TileIndex:InconsistentProjections');
     }
+    if (roundedGsds.size > 1) {
+      if (args.validate) {
+        logger.error({ gsds: [...gsds], roundedGsds: [...roundedGsds] }, 'TileIndex:InconsistentGSDs:Failed');
+        throw new Error(
+          `Inconsistent GSDs found: ${[...roundedGsds].join(', ')} ${[...gsds].join(',')}, ${tiffLocationsToLoad.map(protocolAwareString).join(',')}`,
+        );
+      }
+      logger.warn({ gsds: [...gsds], roundedGsds: [...roundedGsds] }, 'TileIndex:InconsistentGSDs:Failed');
```

Comment:
Noting as discussed it would be good to alert on this or explore setting exit status so that users can be notified if there is a GSD mismatch.